app/src/functions/notify-endpoint-usage/app.ts

```diff
@@ -173,18 +176,46 @@ async function getLoadBalancerThreshold(appData: ApplicationData[], dbApps: IApp
 exports.handler = async () => {
   await connect()
 
-  const usage = await getUsageData();
+  const usage = await retryEvery(getUsageData);
+
+  let networkData: Application[]
+
+  const cachedNetworkData = await redis.get('nt-network-apps')
+  if (!cachedNetworkData) {
+    networkData = await retryEvery(getAppsInNetwork)
+    await redis.set('nt-network-apps', JSON.stringify(networkData, (_, value) =>
+      typeof value === 'bigint'
+        ? value.toString()
+        : value // return everything else unchanged
+    ), 'EX', 3600)
```

Comment:
I'd also lift the redis cache times as we do on the portal backend, into something like `NETWORK_CACHE_TTL`.